packages/mediapipe-task-text/example/lib/text_embedding_demo.dart

```diff
@@ -0,0 +1,471 @@
+
+  final double similarity;
+
+  @override
+  Widget build(BuildContext context) => Text('Similarity score: $similarity');
+}
+
+/// Shows, in the activity feed, a visualiation of an [Embedding] object.
+class TextEmbedderResultDisplay extends StatelessWidget {
+  const TextEmbedderResultDisplay({
+    super.key,
+    required this.embeddedText,
+    required this.index,
+  });
+
+  final TextWithEmbedding embeddedText;
+  final int index;
+
+  @override
+  Widget build(BuildContext context) {
+    if (embeddedText.result == null) {
+      return const CircularProgressIndicator.adaptive();
+    }
+    final embedding = embeddedText.result!.embeddings.last;
+    late String embeddingDisplay;
+    switch (embedding.type) {
+      case EmbeddingType.float:
+        {
+          embeddingDisplay = '${embedding.floatEmbedding!}';
+        }
+      case EmbeddingType.quantized:
+        {
+          embeddingDisplay = '${embedding.quantizedEmbedding!}';
+        }
+    }
+    // Replace "..." with the results
+    final message = '"${embeddedText.value}"\n$embeddingDisplay';
+    return Card(
+      key: Key('Embedding::"${embeddedText.value}" $index'),
+      margin: const EdgeInsets.all(10),
+      child: Padding(
+        padding: const EdgeInsets.all(10),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            Text(message),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Row(
+                  children: <Widget>[
+                    if (embedding.type == EmbeddingType.float)
+                      Padding(
+                        padding: const EdgeInsets.only(right: 8),
+                        child: GFButton(
+                          onPressed: null,
+                          text: 'Float',
+                          shape: GFButtonShape.pills,
+                          color: Colors.blue[300]!,
+                        ),
+                      ),
+                    if (embedding.type == EmbeddingType.quantized)
+                      Padding(
+                        padding: const EdgeInsets.only(right: 8),
+                        child: GFButton(
+                          onPressed: null,
+                          text: 'Quantized',
+                          shape: GFButtonShape.pills,
+                          color: Colors.orange[300]!,
+                        ),
+                      ),
+                    if (embeddedText.l2Normalized)
+                      Padding(
+                        padding: const EdgeInsets.only(right: 8),
+                        child: GFButton(
+                          onPressed: null,
+                          text: 'L2 Normalized',
+                          shape: GFButtonShape.pills,
+                          color: Colors.green[300]!,
+                        ),
+                      ),
+                  ],
+                ),
```

Comment:
Can you swap out the inline ifs for a _getButtonFor(...)? Should de-dupe some of this code and help with the indentation.

Reply:
Done :)